.github/workflows/health_checks.yml

```diff
@@ -43,6 +43,37 @@ jobs:
       - uses: ./.github/actions/restore_build_cache
       - run: npm run set-script-shell
       - run: npm run test:coverage:threshold
+  test_scripts:
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
+      - uses: ./.github/actions/setup_node
+      - uses: ./.github/actions/restore_build_cache
+      - run: npm run set-script-shell
+      - run: npm run test:scripts
+  check_api_changes:
+    if: github.event_name == 'pull_request'
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout pull request ref
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
+      - uses: ./.github/actions/setup_node
+      - uses: ./.github/actions/restore_build_cache
+      - name: Publish packages locally
+        run: npm run vend
+      - name: Checkout base branch
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
+        with:
+          path: base-branch-content
+          ref: ${{ github.event.pull_request.base.sha }}
+      - name: Check API changes
+        run: |
+          mkdir api-validation-projects
```

Comment:
could the script create a temp dir for validation projects internally? This would remove one more setup step (albiet a simple one)

Reply:
discussed offline.
while working on this PR I discovered that it's useful to force user to know where working dir is by making parameter mandatory.
The script intentionally leaves generated projects in place and does not delete them after the run. This gives opportunity to inspect generated code and play with this to understand what broke (e.g. one can open this dir in IDE to further understand the break).